webpack/assets/javascripts/react_app/components/ToastsList/index.js

```diff
@@ -1,9 +1,64 @@
-import { connect } from 'react-redux';
-import * as ToastActions from '../../redux/actions/toasts';
-import ToastsList from './ToastList';
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
 
-const mapStateToProps = state => ({
-  messages: state.toasts.messages,
-});
+import {
+  Alert,
+  AlertActionCloseButton,
+  AlertActionLink,
+  AlertGroup,
+} from '@patternfly/react-core';
 
-export default connect(mapStateToProps, ToastActions)(ToastsList);
+import { addToast, deleteToast } from '../../redux/actions/toasts';
+import { selectToastsListMessages } from './ToastsListSelectors';
+import { toastType, toastTitle } from './ToastListHelpers';
+import './ToastList.scss';
+
+const ToastsList = ({ railsMessages }) => {
+  const dispatch = useDispatch();
+  const messages = useSelector(selectToastsListMessages);
+
+  useEffect(() => {
+    railsMessages.forEach(({ message, type, key }) => {
+      dispatch(addToast({ message, type, key }));
+    });
+  }, [dispatch, railsMessages]);
+
+  const toastsList = Object.entries(messages)
+    .map(([key, message]) => ({ key, ...message }))
+    .map(({ key, type, title, message, link, sticky, ...toastProps }) => (
```

Comment:
I know it's a copy from the older file,
though can we do something like combining the `railsMessages` into the `toastEntries` that are received from the state? so we won't need the `useEffect` and dispatch in order to get it later again from the store? we already have the data in this component, so it feels a bit weird to dispatch it to the store, but maybe I am missing something.

it might also fix the issue you faced

Reply:
How this should behave in  the case of deleting the toast? If I delete `useEffect()`, `railsMessages` won't be dispatched to store so `deleteToast()` won't work on them. This will bring additional logic how to delete messages from `props` & from `redux`

Reply:
alright, that's what I missed :D
though I wonder if the close functionality would be internally managed,
e.g:

```js
actionClose={
  <AlertActionCloseButton onClose={toggleAlert} />
}
```

which might make to component simpler,
and on the top of the component we can have:
```js
const [isVisible, setVisible] = useState(false)
```

although it can be on a follow-up PR, non-blocking this one.